spark/sql/types/arrow.go

```diff
@@ -68,6 +68,12 @@ func ReadArrowTableToRows(table arrow.Table) ([]Row, error) {
 	return result, nil
 }
 
+func ReadArrowRecordToRows(record arrow.Record) ([]Row, error) {
+	table := array.NewTableFromRecords(record.Schema(), []arrow.Record{record})
+	defer table.Release()
+	return ReadArrowTableToRows(table)
+}
+
```

Comment:
Seems a bit inefficient to create a whole array.NewTableFromRecords. We should be able to go straight from records to rows, but I've left this for convenience sake.